doc/articles/Concepts/Authentication.md

```diff
@@ -16,67 +16,183 @@ When you have a multiplayer game, often you need to store information about your
+    {
+        public override void ServerAuthenticate(NetworkConnection conn)
+        {
+            conn.isAuthenticated = true;
+            OnServerAuthenticated.Invoke(conn);
+        }
+
+        public override void ClientAuthenticate(NetworkConnection conn)
+        {
+            conn.isAuthenticated = true;
+            OnClientAuthenticated.Invoke(conn);
+        }
+    }
+}
+```
+
+## Custom Authenticators
+
+Similar to the implementation of Transports in Mirror, the base Authenticator is an abstract class, so to make your own custom Authenticator, you can just create a new script in your project (not in the Mirror folders) that inherits from Authenticator and implements the minimal requirements:
+
+-   You **must** implement **both** `ServerAuthenticate` and `ClientAuthenticate` as overrides, even if one of them might be empty, as in the Basic Authenticator example below.
+
+-   When a client is authenticated to your satisfaction, you **must** set the `isAuthenticated` flag on the `NetworkConnection` to true on **both** the server and client 
+
+-   When a client is authenticated to your satisfaction, you **must** invoke the `OnServerAuthenticated` and `OnClientAuthenticated` events on **both** the server and client
+
+In addition to these requirements, we also *suggest* you do the following:
+
+-   Send a message to the client if authentication fails, especially if there's some issue they can resolve.
+
+-   Call the `Disconnect()` method of the `NetworkConnection` on the server and client when authentication fails.  If you want to give the user a few tries to get their credentials right, you certainly can, but Mirror will not do the disconnect for you.
+
+    -   Remember to put a small delay on the Disconnect call on the server if you send them a failure message so that it has a chance to be delivered before the connection is dropped.
+
+Now that you have the foundation of a custom Authenticator component, the rest is up to you.  You can exchange any number of custom message between the server and client as necessary to complete your authentication process before approving the client.
+
+## Basic Authentication
+
+To get you started, here's a complete example of a custom Authenticator.
+
+-   Create a new script in your project (not in the Mirror folders) called `BasicAuthenticator`
+
+-   Replace the boilerplate code that Unity provides with the code below and save
+
+-   Drag the script to the inspector of the object in your scene that has Network Manager
+
+-   Assign the Basic Authenticator component to the Authenticator field in Network Manager
+
+-   Remove the Default Authenticator component.
+
+When you're done, it should look like this:
 
-4.  Override the `OnServerAddPlayer` method and validate the user's credential.
+![](BasicAuthentication.PNG)
 
-For example this would be part of your `NetworkManager` class:
+### Basic Authenticator
 
 ```cs
-class MyGameNetworkManager : NetworkManager
+using UnityEngine;
```

Comment:
maybe better as an Example?

Reply:
I'll do an actual example, but it will be in the Examples folder inside the Mirror folder with Mirror.Examples namespace.  I know it's a bit long, but I wanted something in the docs that shows what the doc text talks about and that's ready to use via copy paste.

If we could style the code blocks to max height 20 lines and scollable....... :)